23_1.cpp

```diff
@@ -0,0 +1,59 @@
+#include <functional>
+#include <iostream>
+#include <queue>
+#include <vector>
+
+struct ListNode {
+  int val;
+  ListNode* next;
+  explicit ListNode(int x) : val(x), next(nullptr) {}
+};
+
+struct IdWithValue {
+  int id_;
+  int value_;
+  IdWithValue(int id, int value) : id_(id), value_(value) {}
+  bool operator>(IdWithValue other) const {
+    return value_ > other.value_;
+  }
+};
+

```

Comment:
Also, this is better:
```C++
inline bool operator>(const IdWIthValue& lhs, const IdWithValue& rhs) const {
    return lhs.value > rhs.value;
}
```
Here, `lhs`, `rhs` stands for left/right hand side, respectively.

Reply:
The rule is `define a non member function if you can`